ssl/ssl_lib.c

```diff
@@ -4074,45 +4070,138 @@ int SSL_CTX_set_ct_validation_callback(SSL_CTX *ctx, ct_validation_cb callback,
             TLSEXT_TYPE_signed_certificate_timestamp)) {
         SSLerr(SSL_F_SSL_CTX_SET_CT_VALIDATION_CALLBACK,
                SSL_R_CUSTOM_EXT_HANDLER_ALREADY_INSTALLED);
-        goto err;
+        return 0;
     }
 
     ctx->ct_validation_callback = callback;
     ctx->ct_validation_callback_arg = arg;
-    ret = 1;
-err:
-    return ret;
+    return 1;
 }
 
-ct_validation_cb SSL_get_ct_validation_callback(const SSL *s)
+int SSL_ct_is_enabled(const SSL *s)
 {
-    return s->ct_validation_callback;
+    return s->ct_validation_callback != NULL;
 }
 
-ct_validation_cb SSL_CTX_get_ct_validation_callback(const SSL_CTX *ctx)
+int SSL_CTX_ct_is_enabled(const SSL_CTX *ctx)
 {
-    return ctx->ct_validation_callback;
+    return ctx->ct_validation_callback != NULL;
+}
+
+static void *ssl_ct_arg(int vtype)
```

Comment:
I don't think I've explained myself clearly enough, as my proposed change doesn't require any of the above. I'm proposing:
- [ ] 1. changing `ssl_ct_cb` to be: 
```c
 int ssl_ct_policy_satisfied(const CT_POLICY_EVAL_CTX *ctx, const STACK_OF(SCT) *scts, int vtype)
 ```
- [ ] 2. Deleting `ssl_ct_arg`
- [ ] 3. Adding a pair of functions:
```c
 int ssl_ct_permissive_policy_satisfied(const CT_POLICY_EVAL_CTX *ctx, const STACK_OF(SCT) *scts, void *arg) {
   return ssl_ct_policy_satisified(ctx, scts, SSL_CT_POLICY_PERMISSIVE);
 }

 int ssl_ct_strict_policy_satisfied(const CT_POLICY_EVAL_CTX *ctx, const STACK_OF(SCT) *scts, void *arg) {
   return ssl_ct_policy_satisified(ctx, scts, SSL_CT_POLICY_STRICT);
 }
 ```
- [ ] 4. `SSL_enable_ct` just switches on `vtype` and passes one of the two above functions to `SSL_set_ct_validation_callback`.